ci/test/check-merge-with-target.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# check-merge-with-target.sh — check if merge with target branch succeeds and checks still pass.
+
+set -euo pipefail
+
+. misc/shlib/shlib.bash
+
+git fetch "$BUILDKITE_REPO"
+git config --global user.email "buildkite@materialize.com"
+git config --global user.name "Buildkite"
+git merge origin/"$BUILDKITE_PULL_REQUEST_BASE_BRANCH" --message "Merge"
```

Comment:
This works under the assumption that origin == $BUILDKITE_REPO. So you could add $BUILDKITE_REPO as a remote when it's != origin.

Reply:
What about always adding the remote and using that to keep it simple?

✅ I will try that. Let me know what you think.